src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -5,40 +5,35 @@
 
 public class SalaryInfo {
 
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        StringBuilder getSalaryOutput = new StringBuilder();
+        StringBuilder salaryReportBuilder = new StringBuilder();
         int[] salary = new int[names.length];
-        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
-        final LocalDate dateFromFormated = LocalDate.parse(dateFrom, formatter);
-        final LocalDate dateToFormated = LocalDate.parse(dateTo, formatter);
-        getSalaryOutput.append("Report for period ")
+        LocalDate dateFromFormated = LocalDate.parse(dateFrom, formatter);
+        LocalDate dateToFormated = LocalDate.parse(dateTo, formatter);
+        salaryReportBuilder.append("Report for period ")
                 .append(dateFrom)
                 .append(" - ")
                 .append(dateTo);
-        for (String dataArray : data) {
-            String[] dataSplit = dataArray.split(" ");
-            LocalDate localDate = LocalDate.parse(dataSplit[0], formatter);
-            if (localDate.isAfter(dateFromFormated)
-                    && localDate.isBefore(dateToFormated)
-                    || localDate.isEqual(dateFromFormated)
-                    || localDate.isEqual(dateToFormated)) {
-                for (int j = 0; j < names.length; j++) {
-                    if (names[j].equals(dataSplit[1])) {
-                        salary[j] = salary[j]
-                                + (Integer.parseInt(dataSplit[2])
-                                * Integer.parseInt(dataSplit[3]));
-                        break;
+        for (int i = 0; i < names.length; i++) {
+            for (String dataEntry : data) {
+                String[] dataSplit = dataEntry.split(" ");
+                LocalDate localDate = LocalDate.parse(dataSplit[0], formatter);
+                if (!localDate.isBefore(dateFromFormated) && !localDate.isAfter(dateToFormated)) {
```

Comment:
I would extract the check in the if to a separate method for more readability and clarity, eg. isDataWithinRange()